src/espIdf/tracing/tree/appTraceArchiveTreeDataProvider.ts

```diff
@@ -62,9 +62,11 @@ export class AppTraceArchiveTreeDataProvider
 
   public populateArchiveTree() {
     this.appTraceArchives = Array<AppTraceArchiveItems>(0);
-    const workspace = vscode.workspace.workspaceFolders
-      ? vscode.workspace.workspaceFolders[0].uri.fsPath
-      : "";
+    const workspace =
+      vscode.workspace.workspaceFolders &&
+      vscode.workspace.workspaceFolders.length > 0
+        ? vscode.workspace.workspaceFolders[0].uri.fsPath
+        : "";
```

Comment:
I see we use this kind of ternary statements for workspace / folder getting .... think would be a better idea to extract these into the util function !!